src/parser.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import annotations
+import requests as rq
+import json
+from bs4 import BeautifulSoup as bs
+import configuration as cf
+from transformers import pipeline
+
+question_name = "deepset/roberta-base-squad2"
+question_model = pipeline(
+    "question-answering", model=question_name, tokenizer=question_name
+)
+
+
+translation_name = "Helsinki-NLP/opus-mt-ru-en"
+translation_model = pipeline(
+    "translation", model=translation_name, tokenizer=translation_name
+)
+
+
+def pages_num(soup: bs) -> int:
+    pagination = soup.find("div", class_="PaginationWidget__wrapper-pagination")
+    paginator = pagination.find_all("a")
+    num = 0
+    for data_page in paginator:
+        if num < int(data_page["data-page"]):
+            num = int(data_page["data-page"])
+    return num
+
+
+def parce(to_file: str) -> None:
+    try:
+        f = open(to_file, "w")
+    except IOError as e:
+        return None
+
+    classifier_name = "joeddav/xlm-roberta-large-xnli"
+    classifier = pipeline("zero-shot-classification", model=classifier_name)
+    data: dict[str, dict[str, dict]] = {}
+
+    candidate_labels = [
+        "Модуль памяти",
+        "Видеокарта",
+        "Процессор",
+        "Материнская плата",
+        "Блок питания",
+        "Корпус",
+    ]
+
+    url_template = "https://www.citilink.ru/catalog/"
+    matrix = {
+        cf.MB: "materinskie-platy/",
+        cf.CPU: "processory/",
+        cf.GPU: "videokarty/",
+        cf.RAM: "moduli-pamyati/",
+        cf.PB: "bloki-pitaniya/",
+        cf.BODY: "korpusa/",
+    }
+    for cmnt in matrix.keys():
+        data[cmnt] = {}
+        r = rq.get(url_template + matrix[cmnt], allow_redirects=True)
+        print(r.status_code, r.url)
+        soup = bs(r.text, "html.parser")
+        num = pages_num(soup)
+        for i in range(2):  # range(num):
+            r = rq.get(
+                url_template + matrix[cmnt] + "?p=" + str(i + 1) + "&view_type=list"
+            )
+            print(r.status_code)
+            print(r.url)
```

Comment:
A lot of prints without any comments. If these are debugging lines then remove them. If you need this for logging purposes then:
- Use the default python logger, or I suggest using [`loguru`](https://github.com/Delgan/loguru)
- Add a description to what you are outputting

Reply:
Sorry about that. All cleaned. It was helpful in monitoring parser condition.